airflow/api_fastapi/core_api/routes/ui/structure.py

```diff
@@ -63,4 +63,36 @@ def structure_data(
         "edges": edges,
     }
 
+    if external_dependencies:
+        entry_node_ref = nodes[0] if nodes else None
+        exit_node_ref = nodes[-1] if nodes else None
+
+        start_edges: list[dict] = []
+        end_edges: list[dict] = []
+
+        for dependency_dag_id, dependencies in SerializedDagModel.get_dag_dependencies().items():
+            for dependency in dependencies:
+                if dependency_dag_id != dag_id and dependency.target != dag_id:
+                    continue
+
+                # Add nodes
+                nodes.append(
+                    {
+                        "id": dependency.node_id,
+                        "label": dependency.node_id,
```

Comment:
the node_id is a bad label. There was also `dependency.dependency_id` is that more helpful?

Reply:
Yes. `dependency.dependency_id` is better for the label.